### DISCUSSION TITLE
The mesh type is modified

### DISCUSSION THREAD

**Discussion by sg88888** [neperfepx/neper#848] [Q&A] [answered]
Hello,

First of all thank you for developing such a good software for us to use, here are some of the problems I encountered

I want to make a cylindrical foam bullet and the code I use is as follows

neper -T -n 600 -domain "cylinder(2,1)" -reg 1
neper -M n600-id1.tess -dim 2 -rcl 0.6 -meshface "domtype!=2" -nset none -faset none -format msh,inp -o dim2

![1](https://github.com/neperfepx/neper/assets/121993809/075084c2-032b-49af-90c4-170c638b3aa6)

But the generated unit is a 2D unit. How should I modify it to make it a 3D unit? It will still have the same shape.

Then when I checked the grid I found there were many errors, below is the error picture

![2](https://github.com/neperfepx/neper/assets/121993809/79a5dfc0-ad16-4227-878f-1d25e1fd02e2)

Also, I only want to delete one bottom surface of the cylinder, but I don’t want to delete the other circular surface. How to achieve this?

Sorry for asking you so many questions and hope to get your reply

Finally, I would like to express my gratitude to you again

Thanks！

**Comment by rquey** [comment, marked as answer]
This discussion has been closed as obsolete. Please repost if necessary.
